src/Integration.UnitTests/Helpers/ToolWindowServiceTests.cs

```diff
@@ -157,13 +164,37 @@ private class MyDummyToolWindow : IMyDummyToolWindow
             public const string GuidAsString = "45C7FC01-5569-4FE1-AB13-99F32B840D76";
         }
 
-        private void SetupFindToolWindow(Mock<IServiceProvider> serviceProvider, Mock<IVsUIShell> uiShell, int hrResult, Guid toolWindowId, IVsWindowFrame toolWindowObject)
+        private void SetupFindToolWindow(Mock<IVsUIShell> uiShell, int hrResult, Guid toolWindowId, IVsWindowFrame toolWindowObject)
         {
-            serviceProvider.Setup(x => x.GetService(typeof(SVsUIShell))).Returns(uiShell.Object);
-
             uiShell
                 .Setup(x => x.FindToolWindow((uint)__VSFINDTOOLWIN.FTW_fForceCreate, ref toolWindowId, out toolWindowObject))
                 .Returns(hrResult);
         }
+
+        private IVsUIServiceOperation CreateServiceOperation(IVsUIShell svcToPassToCallback)
+        {
+            var serviceOp = new Mock<IVsUIServiceOperation>();
+
+            // Set up the mock to invoke the operation with the supplied VS service
+            serviceOp.Setup(x => x.Execute<SVsUIShell, IVsUIShell>(It.IsAny<Action<IVsUIShell>>()))
+                .Callback<Action<IVsUIShell>>(op => op(svcToPassToCallback));
+
+            return serviceOp.Object;
+        }
+
+        private IVsUIServiceOperation CreateServiceOperation<V>(IVsUIShell svcToPassToCallback)
```

Comment:
This overload sets up the sync `Func` overload.

I'm thinking about adding some test helper methods in `TestInfrastructure` to help with setting up the various `Execute/ExecuteAsync` overrides, but I'm not sure if it's worth it yet. I'll see when I've fixed a few more of the constructors.